tcms/search/__init__.py

```diff
@@ -55,15 +53,14 @@ def advance_search(request):
     results = _query(plan_form.cleaned_data,
                      run_form.cleaned_data,
                      case_form.cleaned_data,
-                     target)
-    results = order_targets(target, results, data)
-    end = time.time()
-    timecost = round(end - start, 3)
+                     data.get('target'))
+    results = order_targets(data.get('target'), results, data)
+    timecost = round(time.time() - start, 3)
     queries = []
-    for form in all_forms:
+    for form in (plan_form, case_form, run_form):
         queries.append(form.cleaned_data)
     queries = fmt_queries(*queries)
-    queries['Target'] = target
+    queries['Target'] = data.get('target')
```

Comment:
Same comment as above for the `target` variable. Do not introduce more complexity then is already there.